sdk/communication/Azure.Communication.CallingServer/src/AmsDirectRequestHelpers.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+
+namespace Azure.Communication.CallingServer
+{
+    internal static class AmsDirectRequestHelpers
+    {
+        internal static HttpMessage GetHttpMessage(CallingServerClient client, Uri requestEndpoint, RequestMethod requestMethod, HttpRange? rangeHeader = null)
+        {
+            HttpMessage message = client._pipeline.CreateMessage();
+            Request request = message.Request;
+            request.Method = requestMethod;
+            var uri = new RequestUriBuilder();
+            uri.Reset(requestEndpoint);
+
+            request.Uri = uri;
+
+            if (rangeHeader != null)
+            {
+                request.Headers.Add(Constants.HeaderNames.Range, rangeHeader.ToString());
+            }
+
+            // Even if using an external location, we must use the acs resource's information to sign our request.
```

Comment:
Can you clarify this comment a little more? I think it means: "When generating security information, we need to use the ACS resource's information, even if we are not connecting directly to our resource. As in the case of downloading a recording."

Also, can you expand on what "ACS resource's information" means.

Reply:
Also, what is path.Substring(1) that you are taking?

Reply:
I moved this over from the existing download code, but think this is just a check to ensure that "/" is not in front of the hostname. path.Substring(1) will strip the the "/" away and return the string beginning after that index.  I'm guessing it's being done this way because we don't want to strip other "/" characters out of the path.